lib/perl/Genome/Model/Command/Admin/ModelSummary.pm

```diff
@@ -324,12 +326,32 @@ sub status_compare { # http://stackoverflow.com/q/540229
 
 
 sub find_first_nondone_step {
+    my $build = shift;
+
+    # The following is wrapped in a transaction and try to protect against
+    # "corrupt" Workflow::Models.
+    my $tx = UR::Context::Transaction->begin();
+    my $first_nondone_step = try {
+        my $wf = $build->newest_workflow_instance;
+        my $step =  _find_first_nondone_step_impl($wf);
+        $tx->commit();
```

Comment:
Does this trigger the `ERROR: Invalid data for save!` message here instead of later?